pkg/operator/discover/discover.go

```diff
@@ -187,21 +192,81 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage string) erro
 
 }
 
+// ListDevices lists all devices discovered on all nodes or specific node if node name is provided.
 func ListDevices(context *clusterd.Context, namespace, nodeName string) (map[string][]sys.LocalDisk, error) {
 	var devices map[string][]sys.LocalDisk
 	listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", k8sutil.AppAttr, discoverDaemon.AppName)}
+	// wait for device discovery configmaps
+	retryCount := 0
+	retryMax := 30
+	sleepTime := 5
+	for {
+		retryCount++
+		if retryCount > retryMax {
+			return devices, fmt.Errorf("exceeded max retry count waiting for device configmap to appear")
+		}
+
+		if retryCount > 1 {
+			// only sleep after the first time
+			<-time.After(time.Duration(sleepTime) * time.Second)
+		}
+
+		cms, err := context.Clientset.CoreV1().ConfigMaps(namespace).List(listOpts)
+		if err != nil {
+			logger.Warningf("failed to list device configmaps: %v", err)
+			return devices, fmt.Errorf("failed to list device configmaps: %+v", err)
+		}
+		if len(cms.Items) == 0 {
+			logger.Infof("no configmap match, retry #%d", retryCount)
+			continue
+		}
+		devices = make(map[string][]sys.LocalDisk, len(cms.Items))
+		for _, cm := range cms.Items {
+			node := cm.ObjectMeta.Labels[discoverDaemon.NodeAttr]
+			if len(nodeName) > 0 && node != nodeName {
+				continue
+			}
+			deviceJson := cm.Data[discoverDaemon.LocalDiskCMData]
+			logger.Debugf("node %s, device %s", node, deviceJson)
+
+			if len(node) == 0 || len(deviceJson) == 0 {
+				continue
+			}
+			var d []sys.LocalDisk
+			err = json.Unmarshal([]byte(deviceJson), &d)
+			if err != nil {
+				logger.Warningf("failed to unmarshal %s", deviceJson)
+				continue
+			}
+			devices[node] = d
+		}
+		break
+	}
+	logger.Debugf("devices %+v", devices)
```

Comment:
Even though it is just a debug message, could you change it to something more like `discovery found the following devices %+v` (will be easier to search the code for when a user could have problems)